services/scabbard/libscabbard/src/service/shared.rs

```diff
@@ -95,15 +106,84 @@ impl ScabbardShared {
         &self.coordinator_service_id
     }
 
-    pub fn add_batch_to_queue(&mut self, batch: BatchPair) {
+    /// Returns the number of pending batches
+    pub fn pending_batches_count(&self) -> usize {
+        self.batch_queue.len()
+    }
+
+    /// set whether we are accepting new batches
+    pub fn set_accepting_batches(&mut self, accepting: bool) {
+        self.accepting_batches = accepting;
+    }
+
+    /// Check whether the pending batch queue has either gotten too big and back pressure has
+    /// should be enabled.
+    /// If back pressure has been enabled, check to see if the queue has reduced in size enough
+    /// to start accepting batches again.
+    fn update_accepting_batches(&mut self) -> Result<(), ScabbardError> {
+        // only the coordinator should change accepting batches and
+        // back pressure is not supported by V1
+        if !self.is_coordinator() || self.scabbard_version == ScabbardVersion::V1 {
+            return Ok(());
+        };
+
+        if self.accepting_batches {
+            // check if batch queue has grown too large
+            if self.pending_batches_count() >= DEFAULT_PENDING_BATCH_LIMIT {
```

Comment:
Use of wrapper functions -- like pending_batches_count() -- requires the reader to reference the function unnecessarily to see the definition, then retain that definition in mind while reading the code.

Reply:
removed